docker/server/build.gradle

```diff
@@ -0,0 +1,92 @@
+
+plugins {
+    id 'com.bmuschko.docker-remote-api'
+}
+
+def baseMap = [
+    'server-base': 'server',
+    'nltk-base': 'server-nltk',
+    'pytorch-base': 'server-pytorch',
+    'sklearn-base': 'server-sklearn',
+    'tensorflow-base': 'server-tensorflow',
+]
+
+baseMap.keySet().each {base ->
+    evaluationDependsOn Docker.registryProject(base)
+}
+
+evaluationDependsOn ':deephaven-jpy'
+evaluationDependsOn ':deephaven-wheel'
+evaluationDependsOn ':deephaven2-wheel'
+
+configurations {
+    serverApplicationDist
+}
+
+dependencies {
+    serverApplicationDist project(path: ':grpc-api-server-native', configuration: 'applicationDist')
+}
+
+def dockerLicenses = License.createFrom(project).syncDockerLicense().get()
+
+def context = { String base ->
+    project.layout.buildDirectory.dir("${baseMap[base]}/context")
+}
+
+def prepareDockerClosure = { String base ->
+    project.tasks.register("prepareDocker-${baseMap[base]}", Sync) {
+        // TODO(deephaven-core#1596): Remove extra dependencies for build-ci.yml
+        // Currently, GH build-ci.yml calls dockerCreateDockerfile, and relies on buildx; so we need to
+        // make sure that dockerCreateDockerfile has the proper dependencies.
+        // If we change how our build works in the future (so that gradle is responsible for pushing the
+        // image), we can remove this extra dependency.
+        dependsOn Docker.registryTask(project, base)
+
+        // deephaven-jpy.whl
+        def deephavenJpyWheel = project(':deephaven-jpy').tasks.getByName('buildWheel')
+
+        // deephaven.whl
+        def deephavenWheel = project(':deephaven-wheel').tasks.getByName('buildWheel')
+
+        // deephaven2.whl
+        def deephaven2Wheel = project(':deephaven2-wheel').tasks.getByName('buildWheel')
+
+        from 'src/main/common'
+        from "src/main/${baseMap[base]}"
+        from (deephavenJpyWheel.outputs.files) {
+            into 'deephaven-jpy-wheel'
+        }
+        from (deephavenWheel.outputs.files) {
+            into 'deephaven-wheel'
+        }
+        from (deephaven2Wheel.outputs.files) {
+            into 'deephaven2-wheel'
+        }
+        from(configurations.serverApplicationDist) {
+            into 'serverApplicationDist'
+        }
+        from(dockerLicenses.outputs) {
+            into 'licenses'
+        }
+        into context(base)
+    }
+}
+
+def prepareDocker = project.tasks.register('prepareDocker').get()
+
+def buildDockerClosure = { String base ->
+    def prepareDockerBase = prepareDockerClosure(base)
+    prepareDocker.dependsOn prepareDockerBase
+    Docker.registerDockerImage(project, "buildDocker-${baseMap[base]}") {
+        dependsOn prepareDockerBase
+        inputDir.set context(base)
+        inputs.files Docker.registryFiles(project, base)
+        buildArgs.put('BASE', "deephaven/${base}:local-build")
+        buildArgs.put('DEEPHAVEN_VERSION', project.version)
+        images.add("deephaven/${baseMap[base]}:local-build")
```

Comment:
to avoid changing this yet again in a few days, would you consider `deephaven/${baseMap[base]}-netty:local-build` as the docker image template, so that a jetty variant can exist for those who a) want docker (because reasons), but b) don't want to copy each in-transit payload through the websocket and envoy proxies?

Happy to consider another naming convention that achieves the same goal

Reply:
I think we should discuss with group; but I'm generally in support of:

* `deephaven/server`, `deephaven/server-jetty` be our defaults
* `deephaven/server-netty` be our alternative

Ditto pattern for all the extended images.

Alternatively, it *may* be possible to package netty and jetty jars, and have it be a runtime configuration? ie,

```
docker run --rm -it deephaven/server --netty
```

```
docker run --rm -it deephaven/server --jetty
```
?

Reply:
i'd vote for -netty/-jetty suffixes over a flag and expanded classpath, but i also like having deephaven/server be an alias for one or the other, and as long as it runs behind envoy and the ws proxy, it won't matter which it is (i.e. clean upgrade path), so we can kick it down the road.